lib/surface/catalogue/components/prop_input.ex

```diff
@@ -126,15 +126,12 @@ defmodule Surface.Catalogue.Components.PropInput do
   defp css_value_to_string(nil), do: nil
   defp css_value_to_string(value), do: Enum.join(value, " ")
 
-  defp get_choices(prop) do
-    values =
-      prop.opts
-      |> Keyword.get(:values, [])
-      |> Enum.map(&{&1, &1})
+  defp get_choices(%{opts: opts}) do
+    values = for x <- Keyword.get(opts, :values, []), do: {x, x}
 
     cond do
       values == [] -> []
-      prop.opts[:required] -> values
+      opts[:required] -> values
       true -> [{"nil", "__NIL__"} | values]
     end
```

Comment:
I wonder if it's better to just return `[]` if `:values` is a range. This way we'll have a `text input` instead of a `select`. It would be weird to have a `select` with all values of a large range, e.g. `1..1000`. 

Another approach would be to check the number of values, something like using a `select` if the number of items Is `<= 10` and leaving as a `text input` if it's `> 10`. WDYT?

Reply:
its a good point @msaraiva. I had considered it but did not have a good solution, this seems like a reasonable approach. I would possibly make it 20-25 rather than 10 though. Sound good?

Reply:
I'm ok with that. Just make sure you try it out to see if it doesn't affect usability badly. I mean, I'd rather type `16` myself than look for a `16` item in a 20-sized list :)

Reply:
yea i'll see what the UX looks like. my thought was that 10 might be too small. Looking for that sweet spot 🍯

Reply:
Ideally, we could have an editable select similar to https://vue-select.org/ or use a `<datalist>` instead of a `<select>`.